File: app/src/main/java/ru/github/pvtitov/senatapp/http_service/HttpClient.java
package ru.github.pvtitov.senatapp.http_service;

import java.util.List;

import ru.github.pvtitov.senatapp.pojos.Item;
import ru.github.pvtitov.senatapp.pojos.Meeting;
import ru.github.pvtitov.senatapp.pojos.Meetings;

public interface HttpClient {

    void requestMeetingsList();
    void requestMeeting(String id);

    interface HttpResponseListener {
        void onSuccess(Meetings meetings);
        void onSuccess(Meeting meeting);
        void onError(String message);
    }
}

File: app/src/main/java/ru/github/pvtitov/senatapp/http_service/HttpClient.kt
package ru.github.pvtitov.senatapp.http_service

import ru.github.pvtitov.senatapp.pojos.Item
import ru.github.pvtitov.senatapp.pojos.Meeting
import ru.github.pvtitov.senatapp.pojos.Meetings

interface HttpClient {

    fun requestMeetingsList()
    fun requestMeeting(id: String)
    fun setHttpResponseListener(listener: HttpClient.HttpResponseListener)

    interface HttpResponseListener {
        fun onSuccess(meetings: Meetings)
        fun onSuccess(meeting: Meeting)
        fun onError(message: String)
    }
}
